library/manageiq_tag_assignment.py

```diff
@@ -0,0 +1,225 @@
+        'category': 'categories', 'cluster': 'clusters', 'data store': 'data_stores',
+        'group': 'groups', 'resource pool': 'resource_pools', 'service': 'services',
+        'service template': 'service_templates', 'template': 'templates',
+        'tenant': 'tenants', 'user': 'users', 'blueprint': 'blueprints'
+    }
+    actions = {'present': 'assign', 'absent': 'unassign'}
+
+    def __init__(self, module, url, user, password, miq_verify_ssl, ca_bundle_path):
+        self.module   = module
+        self.api_url  = url + '/api'
+        self.user     = user
+        self.password = password
+        self.client   = MiqApi(self.api_url, (self.user, self.password), verify_ssl=miq_verify_ssl, ca_bundle_path=ca_bundle_path)
+        self.changed  = False
+
+    def find_entity_by_name(self, entity_type, entity_name):
+        """ Searches the entity name in ManageIQ.
+
+        Returns:
+            the entity id if it exists in manageiq, None otherwise.
+        """
+        entities_list = getattr(self.client.collections, entity_type)
+        return next((e.id for e in entities_list if e.name == entity_name), None)
+
+    def query_resource_tags(self, resource_type, resource_id):
+        """ Returns the tags assigned to the resource.
+        """
+        try:
+            url = '{api_url}/{resource_type}/{resource_id}/tags?expand=resources'.format(api_url=self.api_url, resource_type=resource_type, resource_id=resource_id)
+            result = self.client.get(url)
+            return result.get('resources', [])
+        except Exception as e:
+            self.module.fail_json(msg="Failed to query {resource_type} tags: {error}".format(resource_type=resource_type, error=e))
+
+    def is_tag_assigned(self, resource_type, resource_id, category_name, tag_name):
+        """Return True if the tag is assigned on the resource, False otherwise.
+        """
+        assigned_tags = self.query_resource_tags(resource_type, resource_id)
+        full_tag_name = '/managed/{category_name}/{tag_name}'.format(category_name=category_name, tag_name=tag_name)
+        return any(tag['name'] == full_tag_name for tag in assigned_tags)
+
+    def execute_action(self, resource_type, resource_id, tags, action):
+        """Executes the action for the resource tag
+        """
+        url = '{api_url}/{resource_type}/{resource_id}/tags'.format(api_url=self.api_url, resource_type=resource_type, resource_id=resource_id)
+        try:
+            result = self.client.post(url, action=action, resources=tags)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to {action} tag: {error}".format(action=action, error=e))
+        for result in result['results']:
+            if result['success']:
+                self.changed = True
+            else:
+                self.module.fail_json(msg="Failed to {action}: {fail_message}".format(action=action, entity=entity, fail_message=result['results'][0]['message']))
```

Comment:
I guess `result['results'][0]` here should be simply `result` or however you rename the `for` loop variable?